rust/headless-client/src/lib.rs

```diff
@@ -581,8 +584,10 @@ pub fn debug_command_setup() -> Result<()> {
     let filter = EnvFilter::builder()
         .with_default_directive(tracing_subscriber::filter::LevelFilter::INFO.into())
         .from_env_lossy();
-    let layer = fmt::layer().with_filter(filter);
-    let subscriber = Registry::default().with(layer);
+    let subscriber = SubscriberBuilder::default()
+        .with_span_events(FmtSpan::CLOSE)
```

Comment:
This will log for every span close? this sounds like it can be a bit too noisy by default

Reply:
I mean for debugging

Reply:
Yeah, as soon as I switch to debug log level, it logs every single `decapsulate` call.

What if I use this instead? https://crates.io/crates/logging_timer
It's a new dependency but then I can opt-in individual functions to timing

Reply:
do we need it in main? I figure we will only want this very sparsely when debugging performance

Reply:
Yeah I'll see if I can get by without it. Hopefully it's easy to find again next time. That also means we wouldn't have even manual regression tests for things like startup time